src/transaction/atomic_replication_helper.cpp

```diff
@@ -245,12 +246,17 @@ namespace cublog
     PGBUF_CLEAR_WATCHER (&m_watcher);
   }
 
-  void
+  int
   atomic_replication_helper::atomic_replication_sequence::atomic_replication_unit::apply_log_redo (THREAD_ENTRY *thread_p,
       log_rv_redo_context &redo_context)
   {
     const int error_code = redo_context.m_reader.set_lsa_and_fetch_page (m_record_lsa, log_reader::fetch_mode::FORCE);
-    assert (error_code == NO_ERROR);
+    if (error_code != NO_ERROR)
+      {
+	logpb_fatal_error (thread_p, true, ARG_FILE_LINE,
```

Comment:
Add more info to the logging: `m_record_lsa`, `m_record_index`, whether the page watcher has been used

Nit: `atomic_replication_unit::apply_log_redo: error reading log page; ....`